lib/internal/test_runner/test.js

```diff
@@ -171,18 +171,18 @@ class Test extends AsyncResource {
 
     if (parent === null) {
       this.concurrency = 1;
-      this.indent = '';
+      this.nesting = 0;
       this.only = testOnlyFlag;
-      this.reporter = new TapStream();
+      this.reporter = new TapStream({ objectMode: hasReporters });
```

Comment:
I am debating whether to split `TapStream` into two classes (one for tap, and one for report objects)